djoser/serializers.py

```diff
@@ -24,6 +24,16 @@ class Meta:
             User.USERNAME_FIELD,
         )
 
+    def update(self, instance, validated_data):
+        email = instance.email
+        with transaction.atomic():
+            instance = super(UserSerializer, self).update(instance, validated_data)
+            if validated_data.get('email') and email != instance.email and settings.get('SEND_ACTIVATION_EMAIL'):
```

Comment:
This code is invalid for `User` model with `User. USERNAME_FIELD = 'email'` because the code inside `if` clause will never execute due to `read_only_fields` on serializer set to `User.USERNAME_FIELD`.

What's more in the use case mentioned in previous sentence the code for resending activation email should be present in `SetUsernameSerializer`.